subsys/nrf_security/src/drivers/cracen/cracenpsa/src/wpa3_sae.c

```diff
@@ -0,0 +1,1036 @@
+{
+	psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+
+	/* seed = H(addr1 | addr2, pw | cnt) */
+	status = cracen_wpa3_sae_setup_hmac(&op->mac_op, op->max_id_min_id,
+					    2 * CRACEN_WPA3_SAE_STA_ID_LEN);
+	if (status != PSA_SUCCESS) {
+		goto exit;
+	}
+
+	status = cracen_mac_update(&op->mac_op, op->password, op->pw_length);
+	if (status != PSA_SUCCESS) {
+		goto exit;
+	}
+
+	status = cracen_mac_update(&op->mac_op, counter, 1);
+	if (status != PSA_SUCCESS) {
+		goto exit;
+	}
+
+	status = cracen_mac_sign_finish(&op->mac_op, seed, CRACEN_WPA3_SAE_MAX_SEED_SIZE, seed_len);
+	if (status != PSA_SUCCESS) {
+		goto exit;
+	}
+
+exit:
+	cracen_mac_abort(&op->mac_op);
+	return status;
+}
+
+/* IEEE802.11, 12.7.1.6.2 */
+static psa_status_t sha256_prf_block(cracen_wpa3_sae_operation_t *op,
+				     const uint8_t *key, size_t key_len,
+				     const uint8_t *label, size_t label_len,
+				     const uint8_t *context, size_t context_len,
+				     uint16_t iteration, uint16_t bit_length,
+				     uint8_t *output,
+				     size_t output_size)
+{
+	psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+	uint8_t cnt[sizeof(uint16_t)]; /* represent encoded 16-bit unsigned integers */
+	uint8_t len[sizeof(uint16_t)]; /* represent encoded 16-bit unsigned integers */
+	size_t length;
+
+	/**
+	 * IEEE802.11, 12.7.1.6.2 assumes there might be multiple iterations
+	 * of this KDF. This would be calculated as follows:
+	 *	iteration = key_len / op->hash_length
+	 */
+
+	/* block number */
+	cnt[0] = (uint8_t)iteration;
+	cnt[1] = (uint8_t)(iteration >> 8);
+
+	/* bit length of the derived key */
+	len[0] = (uint8_t)bit_length;
+	len[1] = (uint8_t)(bit_length >> 8);
+
+	status = cracen_wpa3_sae_setup_hmac(&op->mac_op, key, key_len);
+	if (status != PSA_SUCCESS) {
+		goto exit;
+	}
+
+	status = cracen_mac_update(&op->mac_op, cnt, sizeof(cnt));
+	if (status != PSA_SUCCESS) {
+		goto exit;
+	}
+
+	status = cracen_mac_update(&op->mac_op, label, label_len);
+	if (status != PSA_SUCCESS) {
+		goto exit;
+	}
+
+	status = cracen_mac_update(&op->mac_op, context, context_len);
+	if (status != PSA_SUCCESS) {
+		goto exit;
+	}
+
+	status = cracen_mac_update(&op->mac_op, len, sizeof(len));
+	if (status != PSA_SUCCESS) {
+		goto exit;
+	}
+
+	status = cracen_mac_sign_finish(&op->mac_op, output, output_size, &length);
+	if (status != PSA_SUCCESS) {
+		goto exit;
+	}
+
+	return status;
+
+exit:
+	cracen_mac_abort(&op->mac_op);
+	return status;
+}
+
+static int cracen_ec_pt_calc_y_sqr(const struct sx_pk_ecurve *curve,
+				   const sx_const_op *x,
+				   sx_op *sqr_y)
+{
+	int sx_status = SX_ERR_CORRUPTION_DETECTED;
+	uint8_t tmp[CRACEN_P256_KEY_SIZE];
+	uint8_t tmp_2[CRACEN_P256_KEY_SIZE];
+	const struct sx_pk_cmd_def *cmd_mul = SX_PK_CMD_ODD_MOD_MULT;
+	const struct sx_pk_cmd_def *cmd_add = SX_PK_CMD_MOD_ADD;
+	sx_const_op modulo = {.sz = sx_pk_curve_opsize(curve), .bytes = sx_pk_curve_prime(curve)};
+
+	/* y^2 = x^3 + ax + b = (x^2 + a)x + b */
```

Comment:
This comment is confusing since it does not relate to any specific function. If it is a general description please add some context